### PR TITLE
Convert Logger and LocalizationDisambiguator to/from JSON

### PR DESCRIPTION
**Pull request by jdidion**
* Add conversion to/from JSON to Logger (implicit) and LocalizationDisambiguator (explicit)
* Remove deprecated Logger constructors

### CODE REVIEW

File: api/src/main/resources/application.conf [hunk 1]
```diff
@@ -1,5 +1,5 @@
 dxApi {
-  version = "0.6.1-SNAPSHOT"
+  version = "0.7.1-SNAPSHOT"
```
Comment (r-i-v-a) [inline api/src/main/resources/application.conf:2]: Shouldn't it be 0.7.0-SNAPSHOT? Same for other PR with this version.

File: build.sbt [hunk 2]
```diff
@@ -92,8 +92,8 @@ val protocols = project
 
 lazy val dependencies =
   new {
-    val dxCommonVersion = "0.5.2"
-    val dxApiVersion = "0.5.3-SNAPSHOT"
+    val dxCommonVersion = "0.6.1-SNAPSHOT"
+    val dxApiVersion = "0.6.1-SNAPSHOT"
```
Comment (r-i-v-a) [inline build.sbt:96]: version?

**Review by r-i-v-a — APPROVED** [review]
Minor: question about what we're calling the upcoming version

File: build.sbt (hunk 2 above)
Comment (jdidion) [reply build.sbt]: thanks, fixed

**Review by r-i-v-a — APPROVED** [review]

**Pull request merged**